refapp/SKWebApi/Skills/ChatSkill.cs

```diff
@@ -0,0 +1,285 @@
+
+        /*
+         * There will be two types of collections:
+         * 1. ChatMessages: this collection saves all the raw chat messages.
+         * 2. {timestamp}: each of these collections will only have one chat message whose key is the timestamp.
+         * All chat messages will be saved to both kinds of collections.
+         */
+        await context.Memory.SaveInformationAsync(
+            collection: "ChatMessages",
+            text: message,
+            id: messageIdentifier,
+            cancel: context.CancellationToken
+        );
+        await context.Memory.SaveInformationAsync(
+            collection: messageIdentifier,
+            text: formattedMessage,
+            id: currentTime,
+            cancel: context.CancellationToken
+        );
+    }
+
+    private async IAsyncEnumerable<MemoryQueryResult> GetAllMemoriesAsync(SKContext context)
+    {
+        var allCollections = await context.Memory.GetCollectionsAsync(context.CancellationToken);
+        var allChatMessageCollections = allCollections.Where(collection => collection != "ChatMessages");
+        IList<MemoryQueryResult> allChatMessageMemories = new List<MemoryQueryResult>();
+        try
+        {
+            foreach (var collection in allChatMessageCollections)
+            {
+                var results = await context.Memory.SearchAsync(
+                    collection,
+                    "abc",                  // dummy query
+                    limit: 1,
+                    minRelevanceScore: 0.0, // no relevance required since the collection only has one entry
+                    cancel: context.CancellationToken
+                ).ToListAsync();
+                allChatMessageMemories.Add(results.First());
+            }
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine("Exception while retrieving memories: {0}", e.Message);
+            throw e;
+        }
+
+        foreach (var memory in allChatMessageMemories.OrderBy(memory => memory.Id))
+        {
+            yield return memory;
+        }
+    }
+
+    private async Task<MemoryQueryResult> GetLatestMemoryAsync(SKContext context)
+    {
+        var allMemories = this.GetAllMemoriesAsync(context);
+        return await allMemories.FirstAsync();
+    }
+
+    private CompleteRequestSettings GetChatResponseCompletionSettings()
+    {
+        var completionSettings = new CompleteRequestSettings();
+        completionSettings.MaxTokens = SystemPromptDefaults.ResponseTokenLimit;
+        completionSettings.Temperature = SystemPromptDefaults.ResponseTemperature;
+        completionSettings.TopP = SystemPromptDefaults.ResponseTopP;
+        completionSettings.FrequencyPenalty = SystemPromptDefaults.ResponseFrequencyPenalty;
+        completionSettings.PresencePenalty = SystemPromptDefaults.ResponsePresencePenalty;
+        return completionSettings;
+    }
+
+    private CompleteRequestSettings GetIntentCompletionSettings()
```

Comment:
[nit] `CreateIntentCompletionSettings`, same for `GetChatResponseCompletionSettings`